### PR TITLE
Feat/trusted certs

### PR DESCRIPTION
**Pull request by kavics**

### CODE REVIEW

File: src/SenseNet.Client/RESTCaller.cs [hunk 1]
```diff
@@ -294,6 +249,67 @@ public static HttpWebRequest GetStreamRequest(int id, string version = null, str
 
             return GetRequest(url, server);
         }
+        [Obsolete("Do not use anymore.", true)]
```
Comment (tusmester) [inline src/SenseNet.Client/RESTCaller.cs:252]: What should devs use instead?

File: src/SenseNet.Client/RESTCaller.cs [hunk 2]
```diff
@@ -294,6 +249,67 @@ public static HttpWebRequest GetStreamRequest(int id, string version = null, str
...
+
+            ce.Data["Url"] = requestUrl;
+            ce.Data["Method"] = method?.Method ?? HttpMethod.Get.Method;
+            ce.Data["Body"] = body;
+
+            return ce;
+        }
+        #endregion
+
+        public static Task GetStreamResponseAsync(int contentId, Action<HttpResponseMessage> responseProcessor)
```
Comment (tusmester) [inline src/SenseNet.Client/RESTCaller.cs:291]: Consider adding the usual CancellationToken parameter.

File: src/SenseNet.Client/RESTCaller.cs [hunk 3]
```diff
@@ -554,147 +516,192 @@ private static async Task<Content> UploadInternalAsync(Stream binaryStream, Uplo
...
 
-            // we must not close the stream after this as we need to write the chunk into it in the caller method
-            requestStream = myRequest.GetRequestStream();
+        /* ================================================================================ LOW LEVEL API */
 
-            //write form data values
-            foreach (var kvp in uploadData.ToDictionary())
+        public static HttpContent CreateRequestBody(string json)
+        {
+            // serialize json
```
Comment (tusmester) [inline src/SenseNet.Client/RESTCaller.cs:565]: Why use StreamContent instead of StringContent?

File: src/SenseNet.Client/RESTCaller.cs (hunk 1 above)
Comment (kavics) [reply src/SenseNet.Client/RESTCaller.cs:252]: This is a private method. Will be removed with the public caller.

**Pull request merged**